server/service/live_queries.go

```diff
@@ -104,7 +121,61 @@ func runLiveQueryEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 	return res, nil
 }
 
-func runLiveQuery(ctx context.Context, svc fleet.Service, queryIDs []uint, hostIDs []uint) (
+func runLiveQueryOnHostEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*runLiveQueryOnHostRequest)
+
+	if req.Query == "" {
+		return nil, ctxerr.Wrap(ctx, badRequest("query is required"))
+	}
+
+	host, err := svc.HostLiteByIdentifier(ctx, req.Identifier)
```

Comment:
nit. change if not complicated. otherwise it's still ok.


```suggestion
	host, err := svc.HostByIdentifierLite(ctx, req.Identifier)
```